services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ProtectedPathSpec.groovy

```diff
@@ -909,6 +909,80 @@ class ProtectedPathSpec extends BaseSpecification {
         database.resetCosts()
     }
 
+    def "Able to synchronize rules for a flow with protected path"() {
+        given: "Two active not neighboring switches"
+        def switches = topology.getActiveSwitches()
+        def allLinks = northbound.getAllLinks()
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .findAll { src, dst -> src != dst }.find { Switch src, Switch dst ->
+            allLinks.every { link -> !(link.source.switchId == src.dpId && link.destination.switchId == dst.dpId) }
+        } ?: assumeTrue("No suiting switches found", false)
+
+        and: "Create a flow with protected path"
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.allocateProtectedPath = true
+        flowHelper.addFlow(flow)
+
+        def mainFlowPath = northbound.getFlowPath(flow.id).forwardPath
+        def protectedFlowPath = northbound.getFlowPath(flow.id).protectedPath.forwardPath
+        def commonNodeIds = mainFlowPath*.switchId.intersect(protectedFlowPath*.switchId)
+        Wrappers.wait(WAIT_OFFSET) {
+            verifyRulesOnCommonNodes(commonNodeIds, srcSwitch.dpId, dstSwitch.dpId,
+                    mainFlowPath, protectedFlowPath)
+        }
+
+        def uniqueNodes = protectedFlowPath.findAll { !commonNodeIds.contains(it.switchId) } + mainFlowPath.findAll {
+            !commonNodeIds.contains(it.switchId)
+        }
+        verifyRulesOnUniqueSwitches(uniqueNodes)
```

Comment:
Have questions to `verifyRulesOnCommonNodes` and `verifyRulesOnUniqueSwitches`. Hope to see their implementation in other PRs

Reply:
done
PR https://github.com/telstra/open-kilda/pull/2265